packages/ckeditor5-dev-env/lib/release-tools/tasks/update-ckeditor5-dependencies.js

```diff
@@ -0,0 +1,352 @@
+ * repository that is to be updated, as well as optional `commit` flag, that when set to true, will commit the changes from that path.
+ * @param {Boolean} options.dryRun Prevents the script from committing or changing anything, and instead shows list of files that would be
+ * changed.
+ */
+module.exports = function updateCKEditor5Dependencies( options ) {
+	const totalResult = { found: 0, updated: 0, toCommit: 0, differences: [] };
+	const pathsToCommit = [];
+
+	console.log( '\n📍 ' + chalk.blue( 'Updating CKEditor 5 dependencies...\n' ) );
+
+	for ( const pathToUpdate of options.packages ) {
+		const path = pathToUpdate.path.split( /[/\\]/ ).join( posix.sep );
+
+		console.log( `Looking for packages in '${ chalk.underline( path ) }'...` );
+
+		const result = updateDirectory( options.version, path, options.dryRun );
+
+		totalResult.found += result.found;
+		totalResult.updated += result.updated;
+		totalResult.differences.push( ...result.differences );
+
+		if ( !result.found ) {
+			console.log( 'No files were found.\n' );
+		} else if ( !result.updated ) {
+			console.log( `${ chalk.bold( result.found ) } files were found, but none needed to be updated.\n` );
+		} else {
+			console.log( `Out of ${ chalk.bold( result.found ) } files found, ${ chalk.bold( result.updated ) } were updated.\n` );
+
+			if ( pathToUpdate.commit ) {
+				totalResult.toCommit += result.updated;
+				pathsToCommit.push( path );
+			}
+		}
+	}
+
+	if ( options.dryRun ) {
+		console.log( `⚠️ ${ chalk.yellow( 'DRY RUN mode' ) } ⚠️` );
+		console.log( chalk.yellow( `${ chalk.bold( 'Enter' ) } / ${ chalk.bold( 'Space' ) } - Display next file diff` ) );
+		console.log( chalk.yellow( `            ${ chalk.bold( 'A' ) } - Display diff from all files` ) );
+		console.log( chalk.yellow( `      ${ chalk.bold( 'Esc' ) } / ${ chalk.bold( 'Q' ) } - Exit` ) );
+
+		if ( !totalResult.differences.length ) {
+			console.log( chalk.yellow( 'The script has not changed any files.' ) );
+			process.exit();
+		}
+
+		readline.emitKeypressEvents( process.stdin );
+		process.stdin.setRawMode( true );
+
+		// Instead of a lambda function, this `process.stdin.on` has to take in a named function and have `differences` assigned to it.
+		// This is done so that it can be tested properly, as it is otherwise impossible to pass this array inside tests.
+		processInput.differences = totalResult.differences;
+		process.stdin.on( 'keypress', processInput );
+	} else {
+		if ( pathsToCommit.length ) {
+			console.log( '\n📍 ' + chalk.blue( 'Committing the changes...\n' ) );
+
+			for ( const path of pathsToCommit ) {
+				const execOptions = {
+					stdio: 'inherit',
+					cwd: path
+				};
+
+				console.log( `${ chalk.green( '+' ) } '${ chalk.underline( path ) }'` );
+
+				execSync( `git add ${ path }/*/package.json`, execOptions );
+				execSync( 'git commit -m "Internal: Updated CKEditor 5 packages to the latest version. [skip ci]"', execOptions );
+			}
```

Comment:
The current implementation creates more than a single update commit. `git commit` should be executed outside the loop.